apps/explorer/src/app/components/markets/markets-table.tsx

```diff
@@ -0,0 +1,129 @@
+import type { MarketFieldsFragment } from '@vegaprotocol/market-list';
+import { t } from '@vegaprotocol/react-helpers';
+import type {
+  VegaICellRendererParams,
+  VegaValueGetterParams,
+} from '@vegaprotocol/ui-toolkit';
+import { ButtonLink } from '@vegaprotocol/ui-toolkit';
+import type { AgGridReact } from 'ag-grid-react';
+import { AgGridColumn } from 'ag-grid-react';
+import { AgGridDynamic as AgGrid } from '@vegaprotocol/ui-toolkit';
+import { useRef, useLayoutEffect } from 'react';
+import { BREAKPOINT_MD } from '../../config/breakpoints';
+import { MarketStateMapping } from '@vegaprotocol/types';
+import { useAssetDetailsDialogStore } from '@vegaprotocol/assets';
+import type { RowClickedEvent } from 'ag-grid-community';
+import { Link, useNavigate } from 'react-router-dom';
+
+type MarketsTableProps = {
+  data: MarketFieldsFragment[] | null;
+};
+export const MarketsTable = ({ data }: MarketsTableProps) => {
+  const openAssetDetailsDialog = useAssetDetailsDialogStore(
+    (state) => state.open
+  );
+
+  const navigate = useNavigate();
+
+  const ref = useRef<AgGridReact>(null);
+  const showColumnsOnDesktop = () => {
+    ref.current?.columnApi.setColumnsVisible(
+      ['id', 'state', 'asset'],
+      window.innerWidth > BREAKPOINT_MD
+    );
+  };
+
+  useLayoutEffect(() => {
+    window.addEventListener('resize', showColumnsOnDesktop);
+    return () => {
+      window.removeEventListener('resize', showColumnsOnDesktop);
+    };
+  }, []);
+
+  return (
+    <AgGrid
```

Comment:
**nitpick (non-blocking)**: These ag-grids animate to size when they are first rendered. Is this something we can fix? Happy for it to be a follow up if it's possible and a little fiddly.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/6678/218824648-5eba654a-60d0-4b34-8890-35ae011bcbaa.gif)